File: src/BreakfastProvider.Api/Configuration/EventGridConfig.cs
namespace BreakfastProvider.Api.Configuration;

public class EventGridConfig
{
    public string Subject { get; set; } = string.Empty;
    public string Endpoint { get; set; } = string.Empty;
    public string TopicKey { get; set; } = string.Empty;
    public bool IsEnabled { get; set; } = true;
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/config/EventGridConfig.java
package io.lemonlion.breakfast.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

/** Twin of C# {@code EventGridConfig}: endpoint/key/subject for the EventGrid outbox dispatcher. */
@ConfigurationProperties(prefix = "event-grid")
public class EventGridConfig {

    private boolean enabled = true;
    private String endpoint = "";
    private String key = "";
    private String subject = "breakfast/orders";

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public String getEndpoint() {
        return endpoint;
    }

    public void setEndpoint(String endpoint) {
        this.endpoint = endpoint;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }
}
